streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -300,11 +303,27 @@ void handleLostAll() {
         }
     }
 
+    /**
+     * @return Map from task id to that task's overall lag across all state stores
+     */
+    public Map<TaskId, Integer> getTaskLags() {
+        final Map<TaskId, Integer> taskLags = new HashMap<>();
+
+        for (final TaskId id : tasksOnLocalStorage()) {
+            if (isActive(id)) {
+                taskLags.put(id, ACTIVE_TASK_SENTINEL_LAG);
+            } else {
+                taskLags.put(id, 0);
+            }
+        }
+        return taskLags;
+    }
+
     /**
      * Returns ids of tasks whose states are kept on the local storage. This includes active, standby, and previously
      * assigned but not yet cleaned up tasks
      */
-    public Set<TaskId> tasksOnLocalStorage() {
+    Set<TaskId> tasksOnLocalStorage() {
```

Comment:
This was only ever used to encode the subscription info, which is now all handled by `getTaskLag`

Reply:
This method could actually become private, except for a single test. I'm wondering if we can port that test to use getTaskLags instead. Aside from letting us make this private, that would probably improve our testing coverage, since I suppose that test was intended to unit test this class, meaning it should be testing this class's public API, not internal methods.

Reply:
This method is getting refactored somewhat heavily in the next PR, where we actually collect the offsets and will lay in some heavy testing. Still trying to strike a balance between not writing/fixing up any tests until the end, and spending time on 50 tests per PR that are all rendered useless as soon as I start the next one. 
But, I will tighten up this test